py-rattler/src/record.rs

```diff
@@ -0,0 +1,406 @@
+use std::path::PathBuf;
+
+use pyo3::{
+    exceptions::PyTypeError, intern, pyclass, pymethods, FromPyObject, PyAny, PyErr, PyResult,
+};
+use rattler_conda_types::{
+    package::{IndexJson, PackageFile},
+    PackageRecord, PrefixRecord, RepoDataRecord,
+};
+
+use rattler_digest::{parse_digest_from_hex, Md5, Sha256};
+
+use crate::{
+    error::PyRattlerError, package_name::PyPackageName, prefix_record::PyPrefixPaths,
+    version::PyVersion,
+};
+
+#[pyclass]
+#[repr(transparent)]
+#[derive(Clone)]
+pub struct PyRecord {
+    pub inner: RecordInner,
+}
+
+#[derive(Clone)]
+pub enum RecordInner {
+    Prefix(PrefixRecord),
+    RepoData(RepoDataRecord),
+    Package(PackageRecord),
+}
+
+impl PyRecord {
+    pub fn as_package_record(&self) -> PyResult<&PackageRecord> {
```

Comment:
I think you can remove the `PyResult` from this function since it always succeeds.

Reply:
It's true, but it makes `as_{x}_record` API a little inconsistent.

Reply:
Will change this thing.

Reply:
You could also rename the other functions into try_as_.. since they return a result?